cpp/json_ffi/json_ffi_engine.cc

```diff
@@ -52,32 +52,42 @@ bool JSONFFIEngine::AddRequest(std::string request_json_str, std::string request
 
   // inputs
   // TODO: Apply conv template
-  Array<Data> inputs;
+  Conversation conv_template = this->conv_template_;
+  std::vector<std::pair<std::string,
+                        std::optional<std::vector<std::unordered_map<std::string, std::string>>>>>
+      messages;
```

Comment:
Given the `std::pair<....>` is too complicated here, let's introduce a class Message which has the first field `std::string` and the second field `std::optional<...>`. This is mainly for readability.